ДЗ10 готово

EvgeniiKovalev opened the pull request.

EvgeniiKovalev committed:
ДЗ10 готово

FlameXander commented on src/main/java/ru/otus/java/basic/homeworks/homework7/Box.java:
```diff
@@ -0,0 +1,56 @@
+Выполнение методов должно сопровождаться выводом сообщений в консоль.
+*/
+public class Box {
+    public Color color;
```
Почему поле не приватное?

EvgeniiKovalev replied:
сделал приватным.
Ошибочно хотел обойтись без метода класса по перекраске, чтобы извне пользовватель класса мог перекрашивать меняя поле класса, но как дальше вы написали это нарушение инкапсуляции.

FlameXander commented on src/main/java/ru/otus/java/basic/homeworks/homework7/Box.java:
```diff
@@ -0,0 +1,56 @@
+        return this.isOpened;
+    }
+
+    public void setIsOpened(boolean isOpened) throws Exception {
```
Лучше разбить это на 2 метода open() и close()

EvgeniiKovalev replied:
разбил на 2 метода open() и close()

FlameXander commented on src/main/java/ru/otus/java/basic/homeworks/homework7/Box.java:
```diff
@@ -0,0 +1,56 @@
+        this.color = color;
+    }
+
+    public boolean getIsOpened() {
```
Для булеанов обычно не добавляют get - isOpened()

EvgeniiKovalev replied:
переименовал на isOpened()

FlameXander commented on src/main/java/ru/otus/java/basic/homeworks/homework7/Box.java:
```diff
@@ -0,0 +1,56 @@
+        return this.isEmpty;
+    }
+
+    public void setIsEmpty(boolean isEmpty) throws Exception {
```
Лучше разбить на 2 метода put()/clear()

EvgeniiKovalev replied:
разбил на 2 метода put() и clear()

FlameXander commented on src/main/java/ru/otus/java/basic/homeworks/homework7/User.java:
```diff
@@ -0,0 +1,34 @@
+помощью цикла выведите информацию только о пользователях старше 40 лет.
+ */
+public class User {
+    public int yearOfBirth;
```
Почему поле не приватное? - нарушение инкапсуляции

EvgeniiKovalev replied:
сделал геттер getYearOfBirth()

EvgeniiKovalev committed:
ДЗ10 исправлены замечания:
 - сделал приватным color,
 - setIsOpened разбил на 2 метода open() и close()
 - getIsOpened() переименовал в isOpened()
 - setIsEmpty  разбить на 2 метода put() и clear()
 - yearOfBirth сделал приватным и реализовал геттер getYearOfBirth()
